pkg/azuredisk/nodeserver_v2.go

```diff
@@ -297,7 +297,7 @@ func (d *DriverV2) NodeGetCapabilities(ctx context.Context, req *csi.NodeGetCapa
 
 // NodeGetInfo return info of the node on which this plugin is running
 func (d *DriverV2) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoRequest) (*csi.NodeGetInfoResponse, error) {
-	instances, ok := d.cloud.Instances()
+	instances, ok := d.controllerProvisioner.Cloud.Instances()
```

Comment:
We should directly expose the Azure cloud client from the provisioner - it may not be for Azure after all. We could rename `controllerProvisioner` to `cloudProvisioner` and add a `GetNodeInfo` method to it that would get the node information required by this call.